services/workflows-service/src/common/get-file-extension-with-fallback/get-file-extension-with-fallback.ts

```diff
@@ -0,0 +1,32 @@
+import { getFileExtension } from '@/common/get-file-extension/get-file-extension';
+import mime from 'mime';
+import { getFileType } from '@/common/get-file-type/get-file-type';
+
+/**
+ * Returns file's extension with fallback.
+ * @param file - file's path or buffer
+ * @param fileName - file's name with extension
+ */
+export const getFileExtensionWithFallback = async ({
+  file,
+  fileName,
+}: {
+  file: string | Buffer;
+  fileName?: string;
+}) => {
+  const fileMetadata = await getFileType(file);
+
+  if (fileMetadata?.ext) {
+    return fileMetadata.ext;
+  }
+
+  if (!fileMetadata?.ext && fileName) {
```

Comment:
at line 19 you check this ext is not null,
so this check is redundant you can remove it `!fileMetadata?.ext`

Reply:
This was a readability decision. Will remove it though.